### PR TITLE
[2201.3.x] Add examples for map langlib

### LINKED ISSUE #38909: [Task]: Add examples for `lang.map` langlib functions
### Description

$title

### Describe your task(s)

_No response_

### Related area

-> Other Area

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by dulajdilshan**
## Purpose
$title

Fixes #38909 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by SasinduDilshara — DISMISSED** [review]

File: langlib/lang.map/src/main/ballerina/map.bal [hunk 1]
```diff
@@ -30,6 +30,10 @@ type Type1 any|error;
 
 # Returns number of members of a map.
 #
+# ```
```
[the remark this answers is not in the document]
Comment (dulajdilshan) [reply langlib/lang.map/src/main/ballerina/map.bal:33]: We don't use ballerina anymore right?

File: langlib/lang.map/src/main/ballerina/map.bal [hunk 2]
```diff
@@ -103,6 +137,11 @@ public isolated function forEach(map<Type> m, @isolatedParam function(Type val)
 
 # Selects the members from a map for which a function returns true.
 #
+# ```
+# map<int> marks = {"Carl": 85, "Bob": 50, "Max": 60};
+# marks.filter((m) => m >= 60) ⇒ {"Carl": 85, "Max": 60}
```
Comment (pcnfernando) [inline langlib/lang.map/src/main/ballerina/map.bal:142]:
```suggestion
# marks.filter(m => m >= 60) ⇒ {"Carl": 85, "Max": 60}
```

File: langlib/lang.map/src/main/ballerina/map.bal [hunk 3]
```diff
@@ -113,6 +152,11 @@ public isolated function filter(map<Type> m, @isolatedParam function(Type val) r
 
 # Combines the members of a map using a combining function.
 #
+# ```
+# map<int> marks = {"Carl": 85, "Bob": 50, "Max": 60};
+# marks.reduce(isolated function (int accTot, int next) returns int => accTot + next, 0) ⇒ 195;
```
Comment (pcnfernando) [inline langlib/lang.map/src/main/ballerina/map.bal:157]: As discussed in https://github.com/ballerina-platform/ballerina-lang/pull/39073/files#r1051801231

File: langlib/lang.map/src/main/ballerina/map.bal [hunk 4]
```diff
@@ -56,6 +68,10 @@ public isolated function iterator(map<Type> m) returns object {
 
 # Returns the member of a map with given key.
 #
+# ```
+# {"Carl": 85, "Bob": 50, "Max": 60}.get("Carl") ⇒ 85
+# ```
```
Comment (pcnfernando) [inline langlib/lang.map/src/main/ballerina/map.bal:73]: Panic case?

File: langlib/lang.map/src/main/ballerina/map.bal [hunk 5]
```diff
@@ -128,6 +172,12 @@ public isolated function reduce(map<Type> m, @isolatedParam function(Type1 accum
 
 # Removes a member of a map.
 #
+# ```
+# map<int> marks = {"Carl": 85, "Bob": 50, "Max": 60};
+# marks.remove("Carl") ⇒ 85
+# marks ⇒ {"Bob": 50, "Max": 60}
```
Comment (pcnfernando) [inline langlib/lang.map/src/main/ballerina/map.bal:178]: Panic case?

File: langlib/lang.map/src/main/ballerina/map.bal [hunk 6]
```diff
@@ -144,6 +194,12 @@ public isolated function remove(map<Type> m, string k) returns Type = @java:Meth
 # If parameter `m` has a member with key parameter `k`, it removes and returns it;
 # otherwise it returns `()`.
 #
+# ```
+# map<int> marks = {"Carl": 85, "Bob": 50, "Max": 60};
+# marks.removeIfHasKey("Carl") ⇒ 85
+# marks ⇒ {"Bob": 50, "Max": 60}
```
Comment (pcnfernando) [inline langlib/lang.map/src/main/ballerina/map.bal:200]: () return case?

File: langlib/lang.map/src/main/ballerina/map.bal [hunk 7]
```diff
@@ -154,6 +210,12 @@ public isolated function removeIfHasKey(map<Type> m, string k) returns Type? = @
 
 # Removes all members of a map.
 #
+# ```
+# map<int> marks = {"Carl": 85, "Bob": 50, "Max": 60};
+# marks.removeAll();
+# marks ⇒ {}
```
Comment (pcnfernando) [inline langlib/lang.map/src/main/ballerina/map.bal:216]: Panic case?

File: langlib/lang.map/src/main/ballerina/map.bal (hunk 1 above)
[the remark this answers is not in the document]
Comment (pcnfernando) [reply langlib/lang.map/src/main/ballerina/map.bal:33]: Update in all places

**Comment by pcnfernando** [comment]
@dulajdilshan please check on the build

**Review by MaryamZi — APPROVED** [review]

File: langlib/lang.map/src/main/ballerina/map.bal [hunk 8]
```diff
@@ -59,6 +71,14 @@ public isolated function iterator(map<Type> m) returns object {
 # This for use in a case where it is known that the map has a specific key,
 # and accordingly panics if parameter `m` does not have a member with parameter `k` key.
 #
+# ```ballerina
+# map<int> marks = {"Carl": 85, "Bob": 50, "Max": 60};
+#
+# marks.get("Carl") ⇒ 85
+#
+# makrs.get("John") ⇒ panic
```
Comment (MaryamZi) [inline langlib/lang.map/src/main/ballerina/map.bal:79]:
```suggestion
# marks.get("John") ⇒ panic
```

**Review by MaryamZi — CHANGES_REQUESTED** [review]

File: langlib/lang.map/src/main/ballerina/map.bal [hunk 9]
```diff
@@ -43,6 +47,14 @@ public isolated function length(map<any|error> m) returns int =@java:Method {
 # The function `entries` can be used to iterate over the keys and members together.
 # The function `keys` can be used to iterator over just the keys.
 #
+# ```ballerina
+# map<int> marks = {"Carl": 85, "Bob": 50, "Max": 60};
+# object {
+#     public isolated function next() returns record {|int value;|}?;
+# } iterator = marks.iterator();
```
Comment (MaryamZi) [inline langlib/lang.map/src/main/ballerina/map.bal:54]:
```suggestion
# object {
#     public isolated function next() returns record {|int value;|}?;
# } iterator = {"Carl": 85, "Bob": 50, "Max": 60}.iterator();
```

**Review by MaryamZi — APPROVED** [review]

**Review by pcnfernando — APPROVED** [review]

**Pull request merged**